### ISSUE TITLE
After editing subject, new thread retains same subject in Nylas

### ISSUE THREAD

**Issue by stefaniwhylie** [nylas/N1#2738]
If I change the subject on a thread, it shows up in the gmail web client with the new subject, but retains the same subject in N1

Also, while it does start a new thread, the new subject is the thread that retains the history, while the old thread seems to start a new thread. 

Changed "Follow-Up to Yesterday's Call" to "Payroll - Gusto."  In Gmail, this has only 4 messages, which is accurate, but in N1, it's the thread with 14 mesages. 

N1 0.4.47-3a2a94c
OSX 10.11.3

![screenshot 2016-08-18 14 53 57](https://cloud.githubusercontent.com/assets/15056129/17792244/95747b10-6554-11e6-8fdc-a5beca1d229c.png)
![screenshot 2016-08-18 14 54 31](https://cloud.githubusercontent.com/assets/15056129/17792243/95745a68-6554-11e6-977f-d992dddd6c59.png)

**Comment by jackiehluo** [comment]
Hey! What's the expected behavior for you—we create a new thread with one message under the new subject line?

**Comment by stefaniwhylie** [comment]
Yep!  Expected behavior is to treat it as a new thread, with a new subject
line, with the single email (but of course retain the previous messages
from that email)—similar to what Gmail does.

At the very least the subject in Nylas should be changed to the new one.
